project/src/main/java/com/google/sps/data/Places.java

```diff
@@ -45,6 +47,23 @@
     return ImmutableList.copyOf(mutablePlaces);
   }
 
+  /**
+   * Sorts the given list of Places by score.
+   * @param places the list we want to sort.
+   * @param userLocation the user's loaction, used for scores calculations
+   * @return a new list containing the original list's elements
+   *     arranged by descending order of their scores.
+   */
+  public ImmutableList<Place> scoreSort(ImmutableList<Place> places, LatLng userLocation) {
+    List<Place> mutablePlaces = new ArrayList<>(places);
+    Map<Place, Double> placesScores = new PlacesScorer(places, userLocation).getScores();
```

Comment:
I see you made this function non-`static` but still don't get this as a dependency (or at least some factory, as we discussed).

Reply:
Chagned it as discussed in above comments.